contracts/src/test/kotlin/com/template/contracts/ContractTests.kt

```diff
@@ -13,24 +14,55 @@ class ContractTests {
     private val ledgerServices: MockServices = MockServices(listOf("com.template"),
         TestIdentity(CordaX500Name("Test","Test","US")),
         networkParameters = testNetworkParameters(minimumPlatformVersion = 4))
-    var doctor = TestIdentity(CordaX500Name("Alice", "TestLand", "US"))
+
+    var doctor = TestIdentity(CordaX500Name("Doc", "TestLand", "US"))
     var alice = TestIdentity(CordaX500Name("Alice", "TestLand", "US"))
-    var bob = TestIdentity(CordaX500Name("Alice", "TestLand", "US"))
+    var bob = TestIdentity(CordaX500Name("Bob", "TestLand", "US"))
+
+    val appointmentReferenceState = AvailableAppointmentDate("16-01-2000", doctor.party, alice.party, bob.party)
+    val requestState = AppointmentRequest("16-01-2000", doctor.party, alice.party)
+    val appointmentState = Appointment("16-01-2000", doctor.party, alice.party)
 
     @Test
     fun requestTest() {
-        val stateReference = AvailableAppointmentDate("16-01-2000", doctor.party, alice.party, bob.party)
-        val state = AppointmentRequest("16-01-2000", doctor.party, alice.party)
-
         ledgerServices.ledger {
-            //pass
             transaction {
                 //passing transaction
-                reference(AppointmentRequestContract.ID, stateReference)
-                output(AppointmentRequestContract.ID, state)
+                reference(AppointmentRequestContract.ID, appointmentReferenceState)
+                output(AppointmentRequestContract.ID, requestState)
                 command(alice.publicKey, AppointmentRequestContract.Commands.Create())
                 verifies()
             }
+            transaction {
+                //failing transaction (no reference supplied)
+                output(AppointmentRequestContract.ID, requestState)
+                command(bob.publicKey, AppointmentRequestContract.Commands.Create())
+                fails()
+            }
+        }
+    }
+    @Test
+    fun acceptTest() {
```

Comment:
You could check if transaction fails if requested and referenced dates are the same.